lib/explorer/shared.ex

```diff
@@ -217,6 +217,20 @@ defmodule Explorer.Shared do
   defp type(item, _type) when is_nil(item), do: nil
   defp type(item, _type), do: raise(ArgumentError, "unsupported datatype: #{inspect(item)}")
 
+  @doc """
+  The return dtype for the basic arithmetic operations: add, subtract, multiply, and divide.
+
+  This function assumes that the inputs have already had the highest precision enforced.
+  """
+  def dtype_for_basic_arithmetic(operation, left_dtype, right_dtype) do
```

Comment:
This function seemed maybe out of place?

Reply:
I agree. My suggestion is to move the `cast_to_*` functions here and keep this one where it is used. :)

Reply:
Addressed by: [093e9ce](https://github.com/elixir-explorer/explorer/pull/696/commits/093e9cea167c7418707e81032b2dd754ac32afc3)